main.py

```diff
@@ -4,58 +4,115 @@
                                   help='module to use for the data mapper')
 
-    args = parser.parse_args()
+    # transmit parser
+    transmit_parser = parent_subparsers.add_parser(
+        TRANSMIT, help='Connect to a datasource and exectue a query...')
+
+    # positional arguments
+    transmit_parser.add_argument(
+        'module', choices=stix_shifter.TRANSMISSION_MODULES,
+        help='choose which connection module to use'
+    )
+    transmit_parser.add_argument(
+        'connection',
+        type=str,
+        help='Data source Host'
+    )
+    transmit_parser.add_argument(
+        'configuration',
+        type=str,
+        help='Data source Port'
+    )
+
+    # operation subparser
+    operation_subparser = transmit_parser.add_subparsers(title="operation", dest="operation_command")
+    operation_subparser.add_parser(stix_shifter.PING, help="Pings the data source")
+    query_operation_parser = operation_subparser.add_parser(stix_shifter.QUERY, help="Executes a query on the data source")
+    query_operation_parser.add_argument('query_string', help='native datasource query string')
+    results_operation_parser = operation_subparser.add_parser(stix_shifter.RESULTS, help="Fetches the results of the data source query")
+    results_operation_parser.add_argument('search_id', help='uuid of executed query')
+    results_operation_parser.add_argument('offset', help='offset of results')
+    results_operation_parser.add_argument('length', help='length of results')
+    status_operation_parser = operation_subparser.add_parser(stix_shifter.STATUS, help="Gets the current status of the query")
+    status_operation_parser.add_argument('search_id', help='uuid of executed query')
+    operation_subparser.add_parser(stix_shifter.IS_ASYNC, help='Checks if the query operation is asynchronous')
+
+    args = parent_parser.parse_args()
 
     if args.command is None:
-        parser.print_help(sys.stderr)
+        parent_parser.print_help(sys.stderr)
         sys.exit(1)
 
-    options = json.loads(args.options) if bool(args.options) else {}
-    if args.stix_validator:
-        options['stix_validator'] = args.stix_validator
-    if args.data_mapper:
-        options['data_mapper'] = args.data_mapper
-
     shifter = stix_shifter.StixShifter()
-    result = shifter.translate(
-        args.module, args.translate_type, args.data_source, args.data, options=options)
+
+    if args.command == TRANSLATE:
+        options = json.loads(args.options) if bool(args.options) else {}
+        if args.stix_validator:
+            options['stix_validator'] = args.stix_validator
+        if args.data_mapper:
+            options['data_mapper'] = args.data_mapper
+        result = shifter.translate(
+            args.module, args.translate_type, args.data_source, args.data, options=options)
+    elif args.command == TRANSMIT:
+        result = shifter.transmit(args)
+    else:
+        result = "Must call either 'translate' or 'transmit'."
```

Comment:
I don't think you need to process this page, 
it should be handled automatically by the argparser